.github/workflows/compile-kfp.yml

```diff
@@ -0,0 +1,79 @@
+name: Compile KFP pipelines
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+  push:
+    branches: [ main ]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+
+permissions:
+  contents: read
+  actions: write
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: docling-standard
+            dir: kubeflow-pipelines/docling-standard
+            out: docling_convert_pipeline_compiled.yaml
+          - name: docling-vlm
+            dir: kubeflow-pipelines/docling-vlm
+            out: docling_convert_pipeline_compiled.yaml
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install requirements for ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          # Ensure YAML checker is available (if not already required)
+          python -c "import yaml" 2>/dev/null || pip install pyyaml
+
+      - name: Compile ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          set -euxo pipefail
+          python docling_convert_pipeline.py
+
+          # Verify output exists and is non-empty
+          test -s "${{ matrix.out }}"
+
+          # Parse YAML to ensure validity
```

Comment:
@fabianofranz I'm wondering if it's even worth checking the validity at all as a test. If we're trying to test the validity via a script we're probably better off using something internally to KFP to verify the schema is correct in the YAML. 

At the point what are we even testing though, that the output of `kfp.compiler.Compiler().compile()` if the function executes properly? I'm thinking we get rid of a YAML validity check all together, and just make sure it the YAML generated is the same as the one in the directory.

Reply:
I agree that a separate YAML parse step is redundant. Since we’re already using kfp.compiler.Compiler().compile(), compilation itself validates structure and schema. The real value in CI is checking that the compiled IR matches what’s checked into the repo. I’ll remove the YAML parse step and keep it focused on compile + diff